lib/protocol/collection/item_collection.ts

```diff
@@ -117,23 +117,36 @@ export class ItemCollection extends MutableCollection<SNItem> {
       /** Filtered content type map */
       const filteredCTMap = this.filteredMap[contentType]!;
       const sortedElements = this.sortedMap[contentType]!;
+
+      const previousIndex = filteredCTMap[element.uuid];
+      const previousElement = !isNullOrUndefined(previousIndex) ? sortedElements[previousIndex] : undefined;
+      if (element.errorDecrypting || previousElement?.errorDecrypting) {
+        if (element.errorDecrypting !== previousElement?.errorDecrypting) {
+          /** If errorDecrypting property changes a re-sort is necessary.
+           * If previous value is undefined we don't need to care about it. */
+         typesNeedingResort.add(contentType);
+        }
+         /** If the errorDecrypting property stayed the same, we simply replace the value, as a re-sort
+          is not necessary, and the element will not be usable by forthcoming filters */
+        sortedElements[previousIndex] = element;
```

Comment:
Wait, if new `element` is **not** errored, doesn't that mean it is valid, and we can use it in filters below?

Reply:
I update this to check if the new element is valid. Also added a matrix of all possibilities because there are so many options here :D